cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/Job.java

```diff
@@ -101,8 +102,9 @@ void checkTransition(State nextState) {
   boolean isToBeDeleted();
 
   /**
-   * @return the time at which this job was marked for deletion
+   * @return the time at which this job was marked for deletion, null only if {@link #isToBeDeleted} returns false
    */
-  long getToBeDeletedTimestamp();
+  @Nullable
+  Long getToBeDeletedTimestamp();
```

Comment:
it is not obvious. It can be expired due to time-out. I can be obsolete due to a schedule update. It can be deleted because the schedule or the program it was for was deleted. That is why I named it not after its cause but after its intention: that the job needs to be deleted. This time is when it was marked for deletion.